app/models/product_drive.rb

```diff
@@ -76,15 +77,29 @@ def item_quantities_by_name_and_date(date_range)
     end
   end
 
-  def donation_quantity_by_date(date_range)
-    donations.joins(:line_items)
-      .during(date_range)
-      .sum('line_items.quantity')
+  def donation_quantity_by_date(date_range, item_category_id = nil)
+    if [nil, ''].include? item_category_id
+      donations.joins(:line_items)
+        .during(date_range)
+        .sum('line_items.quantity')
+    else
+      donations.joins(line_items: [:item])
+        .where(item: {item_category_id: item_category_id})
+        .during(date_range)
+        .sum('line_items.quantity')
+    end
```

Comment:
I usually try to build the query rather than copy the lines. So I'd write this thusly:

```ruby
query = donations.during(date_range)
if item_category_id.present?
  query = query.joins(line_items: :item).where(item: {item_category_id: item_category_id})
else
  query = query.joins(:line_items)
end
query.sum('line_items.quantity') 
```

Reply:
Thank you, that's a lot cleaner